src/leveled_sst.erl

```diff
@@ -1450,85 +1368,65 @@ fetch(LedgerKey, Hash,
         lookup_slots(
             StartKey,
             EndKey,
             Summary#summary.index,
-            State#state.filter_fun),
+            FilterFun),
     Self = self(),
     SL = length(Slots),
-
-    ExpandedSlots =
-        case SL of
-            1 ->
-                [Slot] = Slots,
-                case RTrim of
-                    true ->
-                        [{pointer, Self, Slot, StartKey, EndKey}];
-                    false ->
-                        [{pointer, Self, Slot, StartKey, all}]
-                end;
-            N ->
-                {LSlot, MidSlots, RSlot} =
-                    case N of
-                        2 ->
-                            [Slot1, Slot2] = Slots,
-                            {Slot1, [], Slot2};
-                        N ->
-                            [Slot1|_Rest] = Slots,
-                            SlotN = lists:last(Slots),
-                            {Slot1, lists:sublist(Slots, 2, N - 2), SlotN}
-                    end,
-                MidSlotPointers = lists:map(fun(S) ->
-                                                {pointer, Self, S, all, all}
-                                                end,
-                                            MidSlots),
-                case RTrim of
-                    true ->
-                        [{pointer, Self, LSlot, StartKey, all}] ++
-                            MidSlotPointers ++
-                            [{pointer, Self, RSlot, all, EndKey}];
-                    false ->
-                        [{pointer, Self, LSlot, StartKey, all}] ++
-                            MidSlotPointers ++
-                            [{pointer, Self, RSlot, all, all}]
-                end
-        end,
-    {SlotsToFetch, SlotsToPoint} =
-        case ScanWidth of
-            SW when SW >= SL ->
-                {ExpandedSlots, []};
-            _ ->
-                lists:split(ScanWidth, ExpandedSlots)
-        end,
-
-    {NeededBlockIdx, SlotsToFetchBinList} =
-        read_slots(Handle,
-                    SlotsToFetch,
-                    {SegChecker, LowLastMod, State#state.blockindex_cache},
-                    State#state.compression_method,
-                    State#state.index_moddate),
-    {NeededBlockIdx, SlotsToFetchBinList, SlotsToPoint}.
+    case SL of
+        1 ->
+            [Slot] = Slots,
+            case RTrim of
+                true ->
+                    [{pointer, Self, Slot, StartKey, EndKey}];
+                false ->
+                    [{pointer, Self, Slot, StartKey, all}]
+            end;
+        N ->
+            {LSlot, MidSlots, RSlot} =
+                case N of
+                    2 ->
+                        [Slot1, Slot2] = Slots,
+                        {Slot1, [], Slot2};
+                    N ->
+                        [Slot1|_Rest] = Slots,
+                        SlotN = lists:last(Slots),
+                        {Slot1, lists:sublist(Slots, 2, N - 2), SlotN}
```

Comment:
For the record. I tried:
```
   [S1 | Rest] = Slots,
   {S1, lists:droplast(Rest), lists:last(Rest)}
```
but my intuition was wrong, it is sometimes faster, sometimes not.

Reply:
Advantage, one does not need a separate case for 2 elements.